src/sentry/workflow_engine/types.py

```diff
@@ -55,6 +62,10 @@ def bulk_get_query_object(data_sources) -> dict[int, T | None]:
 
 
 class DataConditionHandler(Generic[T]):
+    @abstractproperty
+    def type(self) -> DataConditionHandlerType:
```

Comment:
is it weird that to use this type, we'll have to initialize the handler? i'm finding this with using abstract properties for enforcing data conditions json schemas

Reply:
updated to use the ClassVar - it's nice that i can access it statically, but yeah, the typing isn't great or there needs to be a default value. I'll stick with ClassVar + default for now for consistency and cause it has the best general mixture of type safety and customization.